### PR TITLE
Fix for issue #246: Showcase draws on top of navigation bar

### PR DESCRIPTION
**Pull request by sergiiz**
Simple fix for Lollipop

### CODE REVIEW

**Comment by MaxChinni** [comment]
Does it work on a Lollipop device with physical buttons, too?

**Comment by sergiiz** [comment]
@MaxChinni Thanks, good point, this fix will add extra padding in case when device has hardware buttons, I'll update my fix.

**Comment by sergiiz** [comment]
@MaxChinni so I checked possible options for hardware buttons case - this solution http://stackoverflow.com/questions/16092431/check-for-navigation-bar seems to be not the best one -  not always works properly. Maybe here are some other approaches or I'm missing smth?

**Comment by marbat87** [comment]
Does this fix work also in landscape?

**Comment by sergiiz** [comment]
@marbat87 unfortunately no, thanks for a good point, I hope to update it during weeked

File: library/src/main/java/com/github/amlcurran/showcaseview/ShowcaseView.java [hunk 1]
```diff
@@ -343,6 +340,24 @@ public boolean onTouch(View view, MotionEvent motionEvent) {
...
+            int myOrientation = display.getRotation();
+            switch (myOrientation) {
+                case Surface.ROTATION_0:
+                    showcaseView.setPadding(0, 0, 0, ApiUtils.getSoftButtonsBarSizePort(activity));
+                    break;
+                case Surface.ROTATION_90:
+                    showcaseView.setPadding(0, 0, ApiUtils.getSoftButtonsBarSizeLand(activity), 0);
+                    break;
+                case Surface.ROTATION_180:
+                    showcaseView.setPadding(0, 0, ApiUtils.getSoftButtonsBarSizeLand(activity), 0);
```
Comment (rharter) [inline library/src/main/java/com/github/amlcurran/showcaseview/ShowcaseView.java]: This should be `getSoftButtonsBarSizePort`.

Comment (sergiiz) [inline library/src/main/java/com/github/amlcurran/showcaseview/ShowcaseView.java]: Ok, thanks, so it seems that my approach is still not enough generic.

We can try smth like this:
`if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
    showcaseView.setPadding(0, 0, ApiUtils.getSoftButtonsBarSizeLand(activity), ApiUtils.getSoftButtonsBarSizePort(activity));
}`

**Comment by sergiiz** [comment]
See also this thread: https://github.com/SpecialCyCi/AndroidResideMenu/issues/38
so workaround could be in just adding 
`<item name="android:windowDrawsSystemBarBackgrounds">false</item>`
to your  values-v21/styles.xml

File: library/src/main/java/com/github/amlcurran/showcaseview/ShowcaseView.java (hunk 1 above)
Comment (rharter) [inline library/src/main/java/com/github/amlcurran/showcaseview/ShowcaseView.java]: So I was looking into [onApplyWindowInsetListener](https://developer.android.com/reference/android/view/View.html#setOnApplyWindowInsetsListener(android.view.View.OnApplyWindowInsetsListener)) to the view, but it never gets called.  I'm not entirely sure how to use it yet, but that replaces `fitSystemWindows`.  That might be a good option as you don't have to worry about checking orientation and determining where the nav bar is (bottom on tablet landscape, right on phone landscape).

**Pull request closed without merging**